ngraph/core/src/function.cpp

```diff
@@ -347,6 +356,39 @@ bool Function::visit_attributes(AttributeVisitor& visitor)
     return true;
 }
 
+void Function::add_sinks(const SinkVector& sinks)
```

Comment:
yes, if you call add_sinks, only m_sinks will be updated, the same with delete method

Reply:
It looks a little odd to allow Results to be stored in 2 separate lists.

Reply:
Result as output can be only in m_results. Or you about possibility to add Result in m_sinks with add_sinks() function?
 Let's choose:
- save m_results and m_sinks independently, Result nodes as outputs should be in m_results list, but theoretically we can add Result node to m_sinks with add_sink() and this Result will not be output
- duplicate m_results in m_sinks, synchronise these lists: Result nodes added with add_sinks will be added in both m_sinks and m_results. Then actually  we can delete add/delete results and use only add/delete sinks.
For me both variants have pluses and minuses. I like the 1st variant. What do you think @ilyachur @GlebKazantaev  @jane-intel @itikhono as ngraph experts?

Reply:
From my point of view if we have to vectors for Results and Sinks, it is really confuse me, because Sink (in this PR) is a parent of Result, from my point of view it means that result is a sink, and if it is true, it means that sink vector also should contain results.
Or we need to change the hierarchy and proposal.

Reply:
Ok, if it is confusing then let's include m_results in m_sinks. Will fix it.